src/DotNet.Status.Web.Tests/TelemetryControllerTests.cs

```diff
@@ -0,0 +1,81 @@
+using Xunit;
+using DotNet.Status.Web.Controllers;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Microsoft.AspNetCore.Mvc;
+using Kusto.Ingest;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit.Abstractions;
+using System;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
+
+namespace DotNet.Status.Web.Tests
+{
+    public class TelemetryControllerTests
+    {
+        private readonly ITestOutputHelper _output;
+        private TelemetryController _controller;
+
+        public TelemetryControllerTests(ITestOutputHelper output)
+        {
+            _output = output;            
+        }
+
+        protected async Task SetUp(KustoOptions customOptions = null)
+        {
+            var collection = new ServiceCollection();
+            collection.AddOptions();
+            collection.AddLogging(l => {
+                l.AddProvider(new XUnitLogger(_output));
+            });
+
+            collection.Configure<KustoOptions>(options =>
+            {
+                options.KustoIngestConnectionString = customOptions != null ? customOptions.KustoIngestConnectionString : "fakekustoconnectionstring";
+                options.KustoDatabase = customOptions != null ? customOptions.KustoDatabase : "fakekustodatbase";
+            });
+
+            collection.AddScoped<TelemetryController>();
+
+            var kustoIngestionResult = new Mock<IKustoIngestionResult>().Object;
+            var kustoIngestClientMock = new Mock<IKustoIngestClient>();
+            kustoIngestClientMock.Setup(x => x.IngestFromStreamAsync(It.IsAny<System.IO.Stream>(), It.IsAny<KustoIngestionProperties>(), null))
+                .Returns(Task.FromResult(kustoIngestionResult));
+
+            collection.AddSingleton(kustoIngestionResult);
```

Comment:
I don't think we want to put a "Result" into the DI.  It would be little weird for something to pull in "a result".

Reply:
Yep, I over-DI'd this :)